crates/module-system/module-implementations/sov-evm/src/lib.rs

```diff
@@ -187,52 +192,33 @@ impl<S: Spec> Evm<S> {
         )
     }
 
-    /// Access the Ethereum transaction receipts.
-    pub fn receipts<Accessor: AccessoryStateReaderAndWriter>(
-        &self,
-        state: &mut Accessor,
-    ) -> Vec<Receipt> {
-        self.receipts.collect_infallible(state)
-    }
-
     /// Access the Ethereum transaction receipt by number.
     pub fn receipt<Accessor: AccessoryStateReader>(
         &self,
-        number: u64,
-        state: &mut Accessor,
-    ) -> Receipt {
-        self.receipts
-            .get(number, state)
-            .unwrap_infallible()
-            .expect("Receipt for known transaction must be set")
-    }
-
-    /// Access the Ethereum transactions.
-    pub fn transactions<Accessor: AccessoryStateReaderAndWriter>(
-        &self,
+        index: u64,
         state: &mut Accessor,
-    ) -> Vec<TransactionSignedAndRecovered> {
-        self.transactions.collect_infallible(state)
+    ) -> Option<Receipt> {
+        self.receipts.get(&index, state).unwrap_infallible()
     }
 
     /// Access the Ethereum transaction by number.
     pub fn transaction<Accessor: AccessoryStateReader>(
         &self,
-        number: u64,
+        index: u64,
         state: &mut Accessor,
-    ) -> TransactionSignedAndRecovered {
-        self.transactions
-            .get(number, state)
-            .unwrap_infallible()
-            .expect("Transaction with known hash must be set")
+    ) -> Option<TransactionSignedAndRecovered> {
+        self.transactions.get(&index, state).unwrap_infallible()
     }
 
     /// Access the Ethereum blocks.
-    pub fn blocks<Accessor: AccessoryStateReaderAndWriter>(
+    pub fn block_numbers<Accessor: AccessoryStateReaderAndWriter>(
         &self,
         state: &mut Accessor,
-    ) -> Vec<SealedBlock> {
-        self.blocks.collect_infallible(state)
+    ) -> RangeInclusive<u64> {
+        self.block_numbers
+            .get(state)
+            .unwrap_infallible()
+            .expect("Block numbers must be set")
```

Comment:
This is set at genesis so is safe, but all the expect/unwraps will be tacked in another issue.